src/utils/SpokeUtils.ts

```diff
@@ -21,35 +21,26 @@ export async function getBlockRangeForDepositId(
   initLow: number,
   initHigh: number,
   maxSearches: number,
-  spokePool: SpokePoolClient
+  spokePool: Contract,
+  deploymentBlock?: number
 ): Promise<{
   low: number;
   high: number;
 }> {
-  // Resolve the deployment block number.
-  const deploymentBlock = spokePool.deploymentBlock;
+  const { chainId } = await spokePool.provider.getNetwork();
+  deploymentBlock ??= getDeployedBlockNumber("SpokePool", chainId);
+  initHigh = Math.min(initHigh, await spokePool.provider.getBlockNumber());
 
-  // Set the initial high block to the most recent block number or the initial high block, whichever is smaller.
-  initHigh = Math.min(initHigh, spokePool.latestBlockNumber);
-
-  // We will now set a list of sanity checks to ensure that the binary search will not fail
-  // due to invalid input parameters.
-  // If any of these sanity checks fail, then we will throw an error.
+  // We will now set a list of sanity checks to ensure that the binary search will not fail due
+  // to invalid input parameters. If any of these sanity checks fail, then we will throw an error.
   (
     [
-      // Sanity check to ensure that the spoke pool client is updated
-      [spokePool.isUpdated, "Spoke pool client is not updated"],
-      // Sanity check to ensure that initHigh is greater than or equal to initLow.
       [initLow <= initHigh, "Binary search failed because low > high"],
-      // Sanity check to ensure that init Low is greater than or equal to zero.
       [initLow >= deploymentBlock, "Binary search failed because low must be >= deploymentBlock"],
-      // Sanity check to ensure that maxSearches is greater than zero.
       [maxSearches > 0, "maxSearches must be > 0"],
-      // Sanity check to ensure that deploymentBlock is greater than or equal to zero.
       [deploymentBlock >= 0, "deploymentBlock must be >= 0"],
     ] as [boolean, string][]
   ).forEach(([condition, errorMessage]) => {
-    // If the condition is false, then we will throw an error.
```

Comment:
These comments were just 100% mirroring what the code does (and what the error messages said the code was doing), so I opportunistically deleted them but wanted to flag it just in case.